lib/shopify_app/configuration.rb

```diff
@@ -50,6 +53,9 @@ def initialize
       @webhooks_manager_queue_name = Rails.application.config.active_job.queue_name
       @disable_webpacker = ENV['SHOPIFY_APP_DISABLE_WEBPACKER'].present?
       @allow_cookie_authentication = true
+
+      @shop_access_scopes_strategy = ShopifyApp::AccessScopes::ShopStrategy
```

Comment:
I think you probably do need two strategies. You need a noop strategy if the developer isn't storing access scopes (so would always return `false` for `update_access_scopes?`) and the current strategy for developers who are storing them.

So you will need a configuration setting for `redirect_if_scopes_dont_match` or something (naming is hard).

Reply:
I like this idea and as discussed, it'll make sure that the strategies aren't partially turned on.